packages/vuepress-theme-odyssey/components/DocsSidebar.vue

```diff
@@ -0,0 +1,150 @@
+            <DocsLink class="docs-site-title" href="/">{{ title }}</DocsLink>
+            <fieldset v-if="showSearch" class="ods-fieldset">
+              <div class="ods-fieldset-flex">
+                <input
+                  id="search"
+                  class="ods-text-input"
+                  type="search"
+                  name="search"
+                  autocomplete="search"
+                  spellcheck="false"
+                  placeholder="Search"
+                  required
+                />
+                <label class="ods-label" for="search">Search</label>
+              </div>
+            </fieldset>
+          </div>
+          <div
+            :class="{
+              'docs-sidebar--main': true,
+              'is-overflowing': isOverflowing
+            }"
+          >
+            <div ref="mainContent" class="docs-sidebar--main-content">
+              <DocsNav variant="primary" :nav="nav.primary" />
+            </div>
+          </div>
+          <div class="docs-sidebar--footer">
+            <DocsNav variant="secondary" :nav="nav.secondary" />
+          </div>
+        </div>
+      </div>
+    </FocusTrap>
+  </Fragment>
+</template>
+
+<script>
+import { Fragment } from "vue-fragment";
+import { FocusTrap } from "focus-trap-vue";
+
+export default {
+  name: "DocsSidebar",
+  components: {
+    Fragment,
+    FocusTrap,
+    DocsLink: () => import("./DocsLink.vue"),
+    DocsNav: () => import("../components/DocsNav.vue"),
+    DocsTopbar: () => import("../components/DocsTopbar.vue")
+  },
+  props: {
+    showSearch: {
+      type: Boolean,
+      default: false
+    },
+    title: {
+      type: String,
+      default: "default"
+    },
+    nav: {
+      type: Object,
+      default: () => ({})
+    }
+  },
+  data: () => ({
+    isOverflowing: null,
+    isExpanded: false
+  }),
+  watch: {
+    isExpanded: function(isExpanded) {
+      const el = document.documentElement;
+      const className = "is-sidebar-expanded";
+
+      if (isExpanded) {
+        el.classList.add(className);
+
+        // TODO: Figure out a better solution here, this
+        // is required because focus isnt always applied to
+        // the close button when the element is expanded due to
+        // `visibility:hidden;` being applied previously.
+        setTimeout(() => this.$refs.closeButton.focus(), 100);
+      } else {
+        el.classList.remove(className);
+      }
+    },
+    $route: function(to, from) {
+      this.setSidebarState(false);
+    }
+  },
+  beforeMount() {
+    this.setSidebarState(false);
+  },
+  mounted() {
+    if ("IntersectionObserver" in window) {
+      const el = this.$refs.mainContent;
+
+      const observer = new IntersectionObserver(
+        (entries, observer) => {
+          entries.forEach(entry => {
+            if (entry.intersectionRatio !== 1) {
+              this.isOverflowing = true;
+            } else {
+              this.isOverflowing = false;
+            }
+          });
```

Comment:
Are you intending `this` to reference the `entry` object or the component's top-level `isOverflowing` variable?

If it is the latter, notice how multiple elements in the `entries` array will **overwrite** the variable.